lib/sinatra/base.rb

```diff
@@ -1089,7 +1089,16 @@ def invoke
 
     # Dispatch a request with error handling.
     def dispatch!
-      @params.merge!(@request.params).each { |key, val| @params[key] = val && force_encoding(val.dup) }
+      @params.merge!(@request.params).each do |key, val|
+        @params[key] =
+          begin
+            val = val.dup
+          rescue TypeError
```

Comment:
What value of `val` causes `#dup` to raise `TypeError`? Played around in `irb` and couldn't come up with anything.

Reply:
Ah, this is code for Ruby 2.3:

```ruby
$ docker run --rm -it ruby:2.3.8-slim-jessie bash
root@f19c6072d969:/# irb
irb(main):001:0> nil.dup
TypeError: can't dup NilClass
	from (irb):1:in `dup'
	from (irb):1
	from /usr/local/bin/irb:11:in `<main>'
```

Reply:
Such as `true`, `false` etc.. It can be reproduced in ruby <= 2.3.
See: https://github.com/ruby/ruby/blob/v2_3_7/object.c#L398